### PR TITLE
add a build/ folder symlink to bin_dist

### PR DESCRIPTION
**Pull request by walterddr**
## Description
symlink the bin folder to build/ in root directory for easier bin-dist.

### CODE REVIEW

**Comment by walterddr** [comment]
any thoughts on this? it would've simplify our document/quickstart a bit. also personally it makes myself debugging binary distribution easier

File: pinot-distribution/pom.xml [hunk 1]
```diff
@@ -262,6 +262,43 @@
...
           </plugin>
+          <!-- Create symlink "build" to the root directory (Unix only) -->
+          <plugin>
+            <groupId>org.codehaus.mojo</groupId>
+            <artifactId>exec-maven-plugin</artifactId>
+            <version>1.5.0</version>
+            <executions>
+              <execution>
+                <id>remove-build-directory</id>
+                <phase>clean</phase>
```
Comment (xiangfu0) [inline pinot-distribution/pom.xml:273]: Will it fail if user doesn't do `mvn clean package`, but `mvn package`? If not, then we don't need this?

**Review by xiangfu0 — APPROVED** [review]
lgtm

File: pinot-distribution/pom.xml (hunk 1 above)
Comment (walterddr) [reply pinot-distribution/pom.xml:273]: i haven't tested this but removing symlinks before recreating it will do no harm anyways. let's keep this

**Comment by walterddr** [comment]
thx @xiangfu0 --> will create another PR to update pinot-docs

File: pinot-distribution/pom.xml [hunk 2]
```diff
@@ -262,6 +262,43 @@
...
+                  <goal>exec</goal>
+                </goals>
+                <configuration>
+                  <executable>ln</executable>
+                  <arguments>
+                    <argument>-sfn</argument>
+                    <argument>${project.basedir}/target/apache-pinot-${project.version}-bin/apache-pinot-${project.version}-bin</argument>
+                    <argument>${project.basedir}/../build</argument>
+                  </arguments>
+                </configuration>
```
Comment (walterddr) [inline pinot-distribution/pom.xml:298]: symlink part

**Pull request merged**